### PR TITLE
feat(ui): Move Incident Rules settings to org level [SEN-987]

### PR DESCRIPTION
**Pull request by billyvg**
Move from project back to organization level and adds a placeholder for Triggers

image

Fixes SEN-987

### CODE REVIEW

File: src/sentry/static/sentry/app/routes.jsx [hunk 1]
```diff
@@ -255,29 +255,29 @@ function routes() {
...
         componentPromise={() =>
-          import(/* webpackChunkName: "ProjectIncidentRules" */ 'app/views/settings/projectIncidentRules')
+          import(/* webpackChunkName: "OrganizationIncidentRules" */ 'app/views/settings/organizationIncidentRules')
         }
         component={errorHandler(LazyLoad)}
       >
         <IndexRoute
           componentPromise={() =>
-            import(/* webpackChunkName: "IncidentRulesList" */ 'app/views/settings/projectIncidentRules/list')
+            import(/* webpackChunkName: "IncidentRulesList" */ 'app/views/settings/organizationIncidentRules/list')
```
Comment (markstory) [inline src/sentry/static/sentry/app/routes.jsx]: Why not `incidentRules` if there aren't going to be project incident rules the prefix doesn't matter that much.

File: src/sentry/static/sentry/app/views/settings/incidentRules/triggers/list.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+              </SecondaryCondition>
+            </Condition>
+            <Actions>
+              <ul>
+                <li>Email members of #team-billing</li>
+              </ul>
+
+              <Button type="default" icon="icon-edit" size="small">
+                Edit
+              </Button>
```
[the remark this answers is not in the document]
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/incidentRules/triggers/list.tsx]: Placeholder

File: src/sentry/static/sentry/app/views/settings/components/forms/type.tsx [hunk 3]
```diff
@@ -23,16 +23,17 @@ export const FieldType = [
 export type FieldValue = any;
 
 type BaseField = {
-  label: React.ReactNode | (() => React.ReactNode);
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/forms/type.tsx]: e.g. "hidden" fields would not have a label

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx [hunk 4]
```diff
@@ -500,7 +500,7 @@ class FormField extends React.Component {
 
     const observedProps = propsToObserver
       .filter(p => typeof this.props[p] === 'function')
-      .map(p => [p, () => this.props[p](model)]);
+      .map(p => [p, () => this.props[p]({...this.props, model})]);
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx]: @EvanPurkhiser do you remember other places where this is used? I couldn't find instances (besides `<ChoiceMapperField>`), including getsentry.

File: src/sentry/api/serializers/models/alert_rule.py [hunk 5]
```diff
@@ -15,8 +15,6 @@ def serialize(self, obj, attrs, user):
             "id": six.text_type(obj.id),
             "name": obj.name,
             "organizationId": six.text_type(obj.organization_id),
-            # TODO: Remove this once we've migrated to org level
-            "projectId": six.text_type(obj.query_subscriptions.first().project_id),
```
Comment (billyvg) [inline src/sentry/api/serializers/models/alert_rule.py]: @wedamija is this ok to remove?

File: src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+                required: true,
+              },
+              {
+                name: 'projects',
+                type: 'select',
+                label: t('Projects'),
+                help: t('Select projects that this rule will apply to'),
+                choices: projects.map(project => [project.id, project.slug]),
+                placeholder: t('All Projects'),
+                multiple: true,
```
[the remark this answers is not in the document]
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm.tsx]: Added https://github.com/getsentry/sentry/pull/14585/files#diff-e1ac00016d925211cda0983c3ddb82b7R47

File: src/sentry/static/sentry/app/views/settings/incidentRules/triggers/list.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+const Label = styled(Cell)`
+  font-size: 1.2em;
+`;
+
+const Condition = styled(Cell)``;
+
+const MainCondition = styled('div')``;
+const SecondaryCondition = styled('div')`
+  font-size: 0.9em;
```
[the remark this answers is not in the document]
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/incidentRules/triggers/list.tsx]: Done

**Comment by billyvg** [comment]
This should correctly use the org level APIs and render incidents and create incidents. Incident rule details + triggers are all very much a WIP.

File: src/sentry/static/sentry/app/views/settings/incidentRules/create.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+    return (
+      <div>
+        <SettingsPageHeader title={t('New Incident Rule')} />
+        <RuleForm orgId={orgId} onSubmitSuccess={this.handleSubmitSuccess} />
+      </div>
+    );
+  }
+}
+export default withProjects(withOrganization(IncidentRulesCreate));
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/incidentRules/create.tsx]: Do you still need `withProjects` here? Are you using it to force a re-render?

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/views/settings/incidentRules/create.tsx (hunk 8 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/incidentRules/create.tsx]: Thanks, don't need withOrganization either

File: src/sentry/api/serializers/models/alert_rule.py (hunk 5 above)
Comment (wedamija) [reply src/sentry/api/serializers/models/alert_rule.py]: Yeah it's fine with me, I assume it'll break the project level pages, but that shouldn't matter too much.

Comment (billyvg) [reply src/sentry/api/serializers/models/alert_rule.py]: The project level pages are being removed in this PR so shouldn't be a problem

**Pull request merged**